lib/matplotlib/mathtext.py

```diff
@@ -2726,48 +2759,93 @@ def subsuper(self, s, loc, toks):
             result = Hlist([vlist])
             return [result]
 
-        # Handle regular sub/superscripts
-        shift_up = nucleus.height - SUBDROP * xHeight
-        if self.is_dropsub(nucleus):
-            shift_down = nucleus.depth + SUBDROP * xHeight
+        # We remove kerning for consistency (otherwise it will compute kerning
+        # based on non-shrinked characters and may put them very close together
+        # when superscripted)
+        # We change the width of the last character to match the advance to
+        # consider some fonts with weird metrics: e.g. stix's f has a width of
+        # 7.75 and a kerning of -4.0 for an advance of 3.72, and we want to put
+        # the superscript at the advance
+        last_char = nucleus
+        if isinstance(nucleus,Hlist):
+            new_children = nucleus.children
+            if len(new_children):
+                # remove last kern
+                if isinstance(new_children[-1],Kern):
+                    new_children = new_children[:-1]
+                last_char = new_children[-1]
+                last_char.width = last_char._metrics.advance
+            # create new Hlist without kerning
+            nucleus = Hlist(new_children, do_kern=False)
         else:
-            shift_down = SUBDROP * xHeight
+            last_char.width = last_char._metrics.advance
+            nucleus = Hlist([nucleus])
+
+        fs = rcParams['mathtext.fontset']
+        # If a custom fontset is used, check if it is Arev Sans, otherwise use
+        # CM parameters.
+        if fs == 'custom':
+            if (rcParams['mathtext.rm'] == 'sans' and
+                    rcParams['font.sans-serif'][0].lower() == 'Arev Sans'.lower()):
+                fs = 'arevsans'
+            else:
+                fs = 'cm'
```

Comment:
For clarity, can you separate this chunk out to its own private function, perhaps called `_get_fontset_name`?